src/app/controllers/followerships_controller.rb

```diff
@@ -0,0 +1,17 @@
+class FollowershipsController < ApplicationController
+  before_action :set_user
+
+  def create
+    current_user.follow(@user.id)
+    redirect_to user_functions_path(@user.username)
+  end
+
+  def destroy
+    current_user.unfollow(@user.id)
+    redirect_to user_functions_path(@user.username)
+  end
+
+  def set_user
+    @user = User.find(params[:user_id]) || raise(ActiveRecord::RecordNotFound)
```

Comment:
we should use `User.from_param` as it find the user by `username` instead of ID, this prevents attacking the URL with id enumeration, we use `username` for all paths that require finding the user, and the same for function `name` instead of function id.
```suggestion
    @user = User.from_param(params[:user_id]) || raise(ActiveRecord::RecordNotFound)
```